llap-client/src/java/org/apache/hadoop/hive/registry/impl/ZkRegistryBase.java

```diff
@@ -353,6 +356,23 @@ protected final void updateServiceRecord(
     }
   }
 
+  protected void ensurePersistentNodePath(ServiceRecord srv) throws IOException {
+    try {
+      String pNodePath = "/" + PATH_JOINER.join(namespace, StringUtils.substringBetween(workersPath, "/", "/"),
```

Comment:
Is it possible to create a unit test for this join? This one line is bit overloaded with functions and results and better to have a validation against it

Reply:
I agree, this code area needs more clarity than hacking something with "workersPath" and joining it to namespace with slashes
an unit test is much better then a code comment in this case